contrib_bots/lib/github_detail.py

```
@@ -0,0 +1,140 @@
+        # Gets the details of an issues or pull request
+
+        # Try to get an issue, try to get a pull if that fails
+        try:
+            r = requests.get(
+                self.GITHUB_ISSUE_URL_TEMPLATE.format(owner=owner, repo=repo, id=number))
+        except requests.exceptions.RequestException as e:
+            logging.exception(e)
+            return
+
+        if r.status_code == 404:
+            try:
+                r = requests.get(
+                    self.GITHUB_PULL_URL_TEMPLATE.format(owner=owner, repo=repo, id=number))
+            except requests.exceptions.RequestException as e:
+                logging.exception(e)
+                return
+
+        if r.status_code != requests.codes.ok:
+            return
+
+        return r.json()
+
+    def get_owner_and_repo(self, issue_pr):
+        owner = issue_pr.group(1)
+        repo = issue_pr.group(2)
+        if owner is None:
+            owner = self.owner
+            if repo is None:
+                repo = self.repo
+        return (owner, repo)
+
+    def handle_message(self, message, client, state_handler):
+        # type: () -> None
+        # Send help message
+        if message['content'] == '@**{}** help'.format(client.full_name):
+            client.send_message(dict(
+                type='stream',
+                to=message['display_recipient'],
+                subject=message['subject'],
+                content=self.usage(),
+            ))
+
+        # Capture owner, repo, id
+        issue_prs = re.finditer(
+            self.HANDLE_MESSAGE_REGEX, message['content'])
+
+        bot_messages = []
+        for issue_pr in issue_prs:
+            owner, repo = self.get_owner_and_repo(issue_pr)
+            print(owner, repo, issue_pr.group(3))
+            details = self.get_details_from_github(owner, repo, issue_pr.group(3))
+            if details is not None:
+                bot_messages.append(self.format_message(details))
+        # Fail silently
```

This fail silently approach bit me when I was testing the bot. I wasn't specifying issue numbers correctly but had no idea what was wrong. Have you considered sending a messaging when you can locate an issue or repo?